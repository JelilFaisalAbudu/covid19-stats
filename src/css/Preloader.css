#loader-wrapper {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 1000;
}

#loader,
#loader::before,
#loader::after {
  border: 3px solid transparent;
  border-radius: 50%;
}

#loader {
  display: block;
  position: relative;
  left: 50%;
  top: 50%;
  width: 150px;
  height: 150px;
  margin: -75px 0 0 -75px;
  border-top-color: #3498db;
  z-index: 1500;
  -webkit-animation: spin 2s linear infinite;
  animation: spin 2s linear infinite;
}

#loader::before {
  content: "";
  position: absolute;
  top: 5px;
  left: 5px;
  right: 5px;
  bottom: 5px;
  border-top-color: #e74c3c;
  -webkit-animation: spin 3s linear infinite;
  animation: spin 3s linear infinite;
}

#loader::after {
  content: "";
  position: absolute;
  top: 15px;
  left: 15px;
  right: 15px;
  bottom: 15px;
  border-top-color: #f9c922;
  -webkit-animation: spin 1s linear infinite;
  animation: spin 1s linear infinite;
}

@-webkit-keyframes spin {
  0% {
    /* Chrome, Opera 15+, Safari 3.1+ */
    -webkit-transform: rotate(0deg);

    /* IE 9 */
    -ms-transform: rotate(0deg);

    /* Firefox 16+, IE 10+, Opera */
    transform: rotate(0deg);
  }

  100% {
    /* Chrome, Opera 15+, Safari 3.1+ */
    -webkit-transform: rotate(360deg);

    /* IE 9 */
    -ms-transform: rotate(360deg);

    /* Firefox 16+, IE 10+, Opera */
    transform: rotate(360deg);
  }
}

@keyframes spin {
  0% {
    /* Chrome, Opera 15+, Safari 3.1+ */
    -webkit-transform: rotate(0deg);

    /* IE 9 */
    -ms-transform: rotate(0deg);

    /* Firefox 16+, IE 10+, Opera */
    transform: rotate(0deg);
  }

  100% {
    /* Chrome, Opera 15+, Safari 3.1+ */
    -webkit-transform: rotate(360deg);

    /* IE 9 */
    -ms-transform: rotate(360deg);

    /* Firefox 16+, IE 10+, Opera */
    transform: rotate(360deg);
  }
}
