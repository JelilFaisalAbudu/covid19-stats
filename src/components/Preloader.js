import '../css/Preloader.css';

const Preloader = () => (
  <div id="loader-wrapper">
    <div id="loader" />
  </div>
);

export default Preloader;
