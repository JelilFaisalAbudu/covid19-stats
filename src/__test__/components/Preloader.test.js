import '@testing-library/jest-dom/extend-expect';
import { shallow } from 'enzyme';
import '../setup/setupTests';
import Preloader from '../../components/Preloader';

const setup = () => shallow(<Preloader />);

describe('Preloader component', () => {
  let component;
  beforeEach(() => {
    component = setup();
  });

  it('finds the preloader with id #loader', () => {
    const element = component.find('#loader');
    expect(element.length).toEqual(1);
  });

  test('renders the app in the page', () => {
    const item = component.find('#loader');
    expect(item).toMatchSnapshot();
  });
});
