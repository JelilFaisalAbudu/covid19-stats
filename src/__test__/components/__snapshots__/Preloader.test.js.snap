// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preloader component renders the app in the page 1`] = `ShallowWrapper {}`;
